src/extension/mbstyle/src/test/java/org/geoserver/community/mbstyle/web/MBStyleEditPageTest.java

```diff
@@ -81,20 +82,14 @@ public void testMbstyleChange() throws Exception {
         tester.executeAjaxEvent("apply", "click");
         tester.assertModelValue("styleForm:context:panel:previewStyleGroup", true);
         assertNotNull(getCatalog().getStyleByName("mbstyle").getSLD());
-        PointSymbolizer ps =
-                (PointSymbolizer)
-                        getCatalog()
-                                .getStyleByName("mbstyle")
-                                .getStyle()
-                                .featureTypeStyles()
-                                .get(0)
-                                .rules()
-                                .get(0)
-                                .symbolizers()
-                                .get(0);
-        ExternalGraphic eg = (ExternalGraphic) ps.getGraphic().graphicalSymbols().get(0);
-        assertEquals(
-                eg.getURI(),
-                "http://localhost:8080/geoserver/styles/mbsprites#icon=${strURLEncode('circle')}&size=${strURLEncode(Interpolate(POP_MAX,0,0.7,40000000,3.7,'numeric'))}");
+        Style style = getCatalog().getStyleByName("mbstyle").getStyle();
+        Symbolizer sym = style.featureTypeStyles().get(0).rules().get(0).symbolizers().get(0);
+
+        assertTrue(sym instanceof PointSymbolizer);
```

Comment:
@tbarsballe there is a significant regression found by this test, to be handled in the geotools code.

It looks like references to a sprite sheet are not being picked or have changed. Still I would like to track this as a separate issue rather than hold up the pull request (which just covers geoserver integration).

Reply:
Reporting issue here: https://osgeo-org.atlassian.net/browse/GEOT-6522

Reply:
The test itself is intended to just make sure geoserver properly updates the cached SLD for non-SLD styles in the resource pool when a style is saved using the UI; it just happens to use a style that contains an icon.
The icon lookup is to verify the conversion worked.

As for the spritesheet reference - for named icons matching the well-known icon names in GeoServer (e.g. 'circle'), mbstyle should be bypassing the spritesheet entirely. I can't recall where exactly that special case was implemented - I'll try and track it down. So, the issue isn't quite that its not finding the spritesheet, since the spritesheet listed in the style is just a dummy value and doesn't exist 

As an aside, commenting out a portion of the test and setting  the test to `@Ignore` is overkill - just use Ignore and preserve the test content.

Reply:
Andrea points out this change was intensional (https://osgeo-org.atlassian.net/browse/GEOT-6443) so I will update this test case to resolve.

Reply:
Right, that makes sense. You should just be able to grab a spritesheet + sprite name from one of the GeoTools MBStyle tests - the appearance of the style doesn't matter here, just that it gets properly translated into an SLD symbol